ql/src/test/queries/clientpositive/vector_struct_in2.q

```diff
@@ -0,0 +1,22 @@
+set hive.fetch.task.conversion=none;
+
+create table test (a string) partitioned by (y string, m string);
+insert into test values ('aa', 2022, 9);
+
+--original bug report
+select * from test where (y=year(date_sub('2022-09-11',4)) and m=month(date_sub('2022-09-11',4))) or (y=year(date_sub('2022-09-11',10)) and m=month(date_sub('2022-09-11',10)) );
+
+set hive.optimize.point.lookup=false;
+set hive.cbo.enable=false;
+
+--embedded expression in struct
+select * from test where (struct(cast(y as int)) IN (struct(2022)));
+
+--first argument of in expression is const struct
+select * from test where (struct(2022) IN (struct(2022)));
+
+set hive.optimize.constant.propagation=false;
```

Comment:
Can you elaborate on why we need to turn off those properties here? It's probably a good idea to add a comment around that in the qtest itself.

Reply:
Constant propagation optimizer optimizes the filter out so the issue is not reproducible if it is turned on. I'm going to add some comments to the test to make these clear.